chains/nomad-ethereum/src/replica.rs

```diff
@@ -238,6 +240,13 @@ where
             double.1.signature.to_vec().into(),
         );
 
+        let gas_price = self
+            .provider
```

Comment:
you should be able to use `contract.client()` to get a reference to a contract's provider, rather than storing a separate arc copy